src/PositionManager.sol

```diff
@@ -147,29 +139,29 @@ contract PositionManager is IPositionManager, ERC721Permit, PoolInitializer, Mul
     }
 
     /// @param params is an encoding of the Currency to close
-    /// @param sender is the msg.sender encoded by the `modifyLiquidities` function before the `unlockCallback`.
-    /// @return an encoding of int256 the balance of the currency being settled by this call
-    function _close(bytes memory params, address sender) internal returns (bytes memory) {
+    /// @return btyes an encoding of int256 the balance of the currency being settled by this call
+    function _close(bytes memory params) internal returns (bytes memory) {
         (Currency currency) = abi.decode(params, (Currency));
         // this address has applied all deltas on behalf of the user/owner
         // it is safe to close this entire delta because of slippage checks throughout the batched calls.
         int256 currencyDelta = poolManager.currencyDelta(address(this), currency);
 
-        // the sender is the payer or receiver
+        // the locker is the payer or receiver
+        address caller = _getLocker();
```

Comment:
with BaseActionsRouter this should be _msgSender() yes?

Reply:
yeah youll implement
```
function _msgSender {
   return _getLocker()
}
```